packages/util/src/errors.ts

```diff
@@ -5,6 +5,18 @@ interface ErrorLike {
   stack: string;
 }
 
+interface WithInnerError {
+  innerError: string | Error;
+}
+
+/**
+ * Gets whether the given error has an innerError.
+ *
+ * @param error The error to be checked for.
+ */
+const isWithInnerError = (error: unknown): error is WithInnerError =>
+  error !== undefined && typeof error === 'object' && 'innerError' in (error as never);
```

Comment:
```suggestion
  error !== null && typeof error === 'object' && 'innerError' in (error as never);
```
`typeof null === "object"`
while
`typeof undefined === "undefined"`